File: NoticeBoard.server/Models/Notice.cs
namespace NoticeBoard.server.Models
{
    public class Notice
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public DateTime DateCreated { get; set; } = DateTime.Now;

        public int PublisherId { get; set; }
        public string PublisherName { get; set; }
    }

    public class NoticeUpdate
    {
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public int PublisherId { get; set; }
        public string PublisherName { get; set; }
    }
}

File: NoticeBoard.client/src/app/models/notice.ts
export interface Notice {
  id: number;
  title: string;
  content: string;
  dateCreated: Date;
  publisherId: number;
  publisherName: string;
}

export interface NoticeUpdate {
  title: string;
  content: string;
  publisherId: number;
  publisherName: string;
}
